apps/src/lib/util/StatsigReporter.js

```diff
@@ -0,0 +1,100 @@
+import logToCloud from '@cdo/apps/logToCloud';
+import {
+  getEnvironment,
+  isProductionEnvironment,
+  isStagingEnvironment,
+  isDevelopmentEnvironment,
+} from '../../utils';
+
+// A flag that can be toggled to send events regardless of environment
+const ALWAYS_SEND = false;
+const Statsig = require('statsig-js');
+
+class StatsigReporter {
+  constructor() {
+    const local_mode = isProductionEnvironment ? false : true;
```

Comment:
Did you mean for this to be `isProductionEnvironment()`?

Reply:
When I was setting up the Ruby SDK, the Infrastructure team emphasized the importance of maintaining consistency between our managed server test environment and production by ensuring Statsig configurations are identical in both environments, I.e. we are setting `local_mode` to false in both those environments. [Here](https://github.com/code-dot-org/code-dot-org/pull/56879/files#diff-f78118b42ccf9fa4ed82455846845c0513a11f72c375070da73aa25531983475R6) is how I did that on the Rails side, to ensure I was targeting the managed server test environment, not other test environments (drone, etc). I'm not sure if there is similar methods we can use in JS?

Reply:
Nick- I saw that code and wasn't quite sure what the front end equivalent would be. I tried to write something that would seem to fit both your Ruby integration, and what we have already in AnalyticsReporter (Amplitude). That said- I'll poke around more. And good catch Sanchit- fixed!

Reply:
@nicklathe in my latest commit, I loaded the same logic into the application.html.haml file so I could query it from the DOM to mimic what you did for the ruby sdk. Let me know if it makes sense/looks right!